File: python/insert_dashes.py
"""ID: 55960bbb182094bc4800007b

Write a function that takes an integer `num` (`num >= 0`) and inserts dashes (`'-'`) between each two **odd** digits in `num`.

## Examples

```python
454793 ---> "4547-9-3"
     0 ---> "0"
     1 ---> "1"
13579  ---> "1-3-5-7-9"
 86420 ---> "86420"
```"""


def is_odd(num: int) -> bool:
    return num % 2 == 1


def insert_dash(num: int) -> str:
    num = str(num)
    new_string = "0"
    for i in num:
        new_string += "-" + i if is_odd(int(i)) and is_odd(int(new_string[-1])) else i
    return new_string[1:]

File: js/insert-dashes.js
/* ID: 55960bbb182094bc4800007b

Write a function that takes an integer `num` (`num >= 0`) and inserts dashes (`'-'`) between each two **odd** digits in `num`.

## Examples

```python
454793 ---> "4547-9-3"
     0 ---> "0"
     1 ---> "1"
13579  ---> "1-3-5-7-9"
 86420 ---> "86420"
``` */

"use strict";


const insertDash = (numbers) => {
     const isOdd = num => num % 2 == 1;

     let newString = "0";
     Array.from(numbers.toString()).forEach(num => {
          if (isOdd(parseInt(num)) && isOdd(parseInt(newString.charAt(newString.length - 1)))) {
               newString += "-" + num
          } else {
               newString += num
          }
     });

     return newString.slice(1);
}
